core/src/util/builders.cpp

```diff
@@ -59,22 +45,72 @@ void Builders::buildPolygon(const Polygon& _polygon, float _height, PolygonBuild
         }
     }
 
-    // call the tesselator
-    glm::vec3 normal(0.0, 0.0, 1.0);
+    _ctx.earcut(_polygon);
+
+    size_t sumPoints = 0;
+    for (auto& line : _polygon) {
+        sumPoints += line.size();
+    }
 
-    _ctx.numVertices += _ctx.earcut.vertices.size();
+    size_t sumVertices = 0;
+    _ctx.used.assign(sumPoints, 0);
+    for (auto i : _ctx.earcut.indices) {
```

Comment:
Ok I see. I'm fine with this implementation but it does need some comments describing the procedure - I for one will certainly forget how this works by the next time we need to look at it.